### DISCUSSION TITLE
Is there a way to "save without formatting?"

### DISCUSSION THREAD

**Discussion by bbuck** [w0rp/ale#4563] [Q&A - Ask for help with problems]
I'm working a project where we have some older JS files with many linting issues -- sometimes fixing these files produces massive PRs for what amounts to single line changes so we have a pattern of saving without formatting and manually fixing issues in certain files.

If I have `let g:ale_fix_on_save = 1` set is there a way to save without formatting for a specific file?

**Comment by w0rp** [comment]
Yes, the way is to set `let b:ale_fixers = []` for particular buffers. I tend to just perform regex matches on file paths in an ftplugin file like so to change the fixers for different files:

```vim
if expand('%:p') =~# 'neural'
    let b:ale_fixers = ['isort']
endif
```

`g:ale_pattern_options` does the same basic thing if you want to control this behaviour in vimrc. Check out `:help g:ale_pattern_options`.

**Reply by bbuck** [reply]
Hmm, I suppose that I was looking for something a bit more off the cuff. I can't easily identify these files, and it's dependent on the size of changes I plan to make in the file. For example if the file is 1k+ LOC and I'm adding 5 lines, it's not worth fixing the entire file in the PR -- but if I'll be rewriting some significant portions of the file then I'll go ahead and and lint.

I'm coming from VS Code, and in VS Code I have Lint on Save set up so if run `Cmd+s` then it lints, but if I don't want to lint a specific file on save I can use the shortcut (may not be default) `Cmd+k s` to save without formatting.
